strawberryfields/backends/gaussianbackend/backend.py

```diff
@@ -27,198 +27,112 @@
 class GaussianBackend(BaseGaussian):
     """Gaussian backend implementation"""
     def __init__(self):
-        """
-        Instantiate a GaussianBackend object.
+        """Initialize the backend.
         """
         super().__init__()
         self._supported["mixed_states"] = True
         self._short_name = "gaussian"
+        self._init_modes = None
+        self.circuit = None
 
     def begin_circuit(self, num_subsystems, *, cutoff_dim=None, pure=None, **kwargs):
-        r"""
-        Create a quantum circuit (initialized in vacuum state) with number of modes
-        equal to num_subsystems.
+        r"""Instantiate a quantum circuit.
+
+        Instantiates a representation of a quantum optical state with num_subsystems modes.
+        The state is initialized to vacuum.
+
+        The modes in the circuit are indexed sequentially using integers, starting from zero.
+        Once an index is assigned to a mode, it can never be re-assigned to another mode.
+        If the mode is deleted its index becomes invalid.
+        An operation acting on an invalid or unassigned mode index raises an IndexError exception.
 
         Args:
-            num_subsystems (int): number of modes the circuit should begin with
+            num_subsystems (int): number of modes in the circuit
+            cutoff_dim (int): unused in this backend
```

Comment:
Do these unused args show up somewhere in the compiled docs, necessitating an extra explainer here?

Reply:
The Fock backends use them, and they appear in the method call signature so some documentation is needed. Pylint frowns upon changing the call signature when overriding a parent class method so they can't just be removed. One solution would be to move them explicitly into kwargs (that's how the Engine already treats them).

Reply:
I'm okay with the solution used above, as it satisfies the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle)

Reply:
I changed them into kwargs, now the docstrings look nice.